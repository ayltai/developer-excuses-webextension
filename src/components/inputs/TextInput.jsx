import { Box, makeStyles, TextField, } from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';

export const TextInput = props => {
    const classes = makeStyles(theme => ({
        margin : {
            marginBottom : theme.spacing(2),
        },
    }))();

    const [ value, setValue, ] = React.useState(props.value);

    return (
        <Box className={classes.margin}>
            <TextField
                label={props.label}
                placeholder={props.placeholder}
                variant='filled'
                multiline={props.multiline}
                rows={props.multiline ? 5 : 1}
                rowsMax={props.multiline ? 5 : 1}
                value={value}
                fullWidth
                onChange={event => {
                    setValue(event.target.value);

                    if (props.onChange) props.onChange(event.target.value);
                }} />
        </Box>
    );
};

TextInput.propTypes = {
    label       : PropTypes.string,
    placeholder : PropTypes.string,
    multiline   : PropTypes.bool,
    value       : PropTypes.string,
    onChange    : PropTypes.func,
};
