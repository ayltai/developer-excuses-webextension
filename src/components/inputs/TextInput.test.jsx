import { TextField, } from '@material-ui/core';
import { mount, shallow, } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import { TextInput, } from './TextInput';

describe('<TextInput />', () => {
    it('renders correctly', () => {
        expect(renderer.create(
            <TextInput
                label='dummy'
                placeholder='dummy'
                value='dummy'
                onChange={() => {}} />
        ).toJSON()).toMatchSnapshot();
    });

    it('mounts without error', () => {
        mount(
            <TextInput
                label='dummy'
                placeholder='dummy'
                multiline
                value='dummy'
                onChange={() => {}} />
        );
    });

    it('triggers onChange when changed', () => {
        const onChange = jest.fn();

        shallow(
            <TextInput
                label='dummy'
                placeholder='dummy'
                multiline
                value='dummy'
                onChange={onChange} />)
            .find(TextField)
            .simulate('change', {
                target : {
                    value : 'dummy',
                },
            });

        expect(onChange.mock.calls.length).toEqual(1);
    });

    it('does not trigger onChange when changed', () => {
        let value = '';

        shallow(
            <TextInput
                label='dummy'
                placeholder='dummy'
                multiline
                value={value} />)
            .find(TextField)
            .simulate('change', {
                target : {
                    value : 'dummy',
                },
            });

        expect(value).toEqual('');
    });
});
