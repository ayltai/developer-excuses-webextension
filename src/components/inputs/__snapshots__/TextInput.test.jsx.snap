// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> renders correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 makeStyles-margin-1"
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-filled"
      data-shrink={true}
    >
      dummy
    </label>
    <div
      className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiFilledInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="dummy"
        required={false}
        rows={1}
        type="text"
        value="dummy"
      />
    </div>
  </div>
</div>
`;
